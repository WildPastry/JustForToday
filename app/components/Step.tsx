import { IStep } from '../types/data.types';
import { StyleSheet } from 'react-native';
import { Text } from './Themed';

const StepItem: React.FC<IStep> = (step: IStep): JSX.Element => {
  return (
    // Step
    <Text style={styles.text}>
      #{step.id}: {step.step}
    </Text>
  );
};

const styles = StyleSheet.create({
  text: {
    textAlign: 'left',
    lineHeight: 20,
    marginBottom: 10
  }
});

export default StepItem;
