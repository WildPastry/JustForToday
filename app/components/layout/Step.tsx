import { IStep } from '../../types/data.types';
import { StyleSheet } from 'react-native';
import { Text } from '../styles/Themed';

const StepItem: React.FC<IStep> = (props: IStep): JSX.Element => {
  return (
    // Step
    <Text style={styles.text}>
      #{props.id}: {props.step}
    </Text>
  );
};

const styles = StyleSheet.create({
  text: {
    textAlign: 'left',
    lineHeight: 25,
    marginBottom: 20
  }
});

export default StepItem;
