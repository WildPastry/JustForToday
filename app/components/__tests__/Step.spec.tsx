import { Provider } from 'react-redux';
import Step from '../Step';
import { makeStore } from '../../redux/store';
import renderer from 'react-test-renderer';

describe('<Step />', () => {
  it('renders correctly', () => {
    const store = makeStore();
    const tree = renderer
      .create(
        <Provider store={store}>
          <Step id={''} step={''} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
