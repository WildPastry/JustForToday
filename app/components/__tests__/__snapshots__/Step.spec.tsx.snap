// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Step /> renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#000",
      },
      undefined,
    ]
  }
>
  #
  : 
</Text>
`;
